userspace/libsinsp/sinsp_ppm_sc.cpp

```diff
@@ -238,6 +251,57 @@ std::unordered_set<uint32_t> sinsp::enforce_sys_ppm_sc_set(std::unordered_set<ui
 	return ppm_sc_set;
 }
 
+std::unordered_set<std::string> erase_io_syscalls_ppm_sc_codes(std::unordered_set<uint32_t>& ppm_sc_of_interest, std::unordered_set<uint32_t>& io_ppm_sc_set)
```

Comment:
@FedeDP would you have more ideas here? It doesn't fit the current sinsp API style, but for Falco we would also need the set of erased syscalls names. Happy to split and adjust depending on what you would prefer the most. Thanks!

Reply:
Intersection, union, and difference operations on `unordered_set`s are starting to become common ops both here and in Falco. I'd be in favor of creating generig utilities/helpers for those as part of the libsinsp utils eventually.

Reply:
why not use directly `std::set` in these APIs? we are at initialization time so probably also `std::set` should be ok :thinking:

Reply:
I think at this point we are stuck with unordered_set and since we need to land the adaptive syscalls selection mechanisms (spanning multiple PRs across both repos), it could be something to be considered for a refactor later? WDYT? In general yes I agree, just the timing seems not right atm.

Reply:
make sense, i would only avoid to introduce new helpers for unordered sets if we don't need them :)